extensions/redis-client/deployment/src/main/java/io/quarkus/redis/client/deployment/RedisClientProcessor.java

```diff
@@ -0,0 +1,38 @@
+package io.quarkus.redis.client.deployment;
+
+import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
+import io.quarkus.deployment.Feature;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.ExtensionSslNativeSupportBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.RuntimeInitializedClassBuildItem;
+import io.quarkus.smallrye.health.deployment.spi.HealthBuildItem;
+import io.vertx.redis.client.impl.types.BulkType;
+
+public class RedisClientProcessor {
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(Feature.REDIS_CLIENT);
+    }
+
+    @BuildStep
+    ExtensionSslNativeSupportBuildItem activateSslNativeSupport() {
+        return new ExtensionSslNativeSupportBuildItem(Feature.REDIS_CLIENT.getName());
+    }
+
+    @BuildStep
+    AdditionalBeanBuildItem registerAPIsProducer() {
+        return AdditionalBeanBuildItem.unremovableOf("io.quarkus.redis.client.runtime.RedisAPIProducer");
+    }
+
+    @BuildStep
+    HealthBuildItem addHealthCheck(RedisBuildTimeConfig buildTimeConfig) {
+        return new HealthBuildItem("io.quarkus.redis.client.health.RedisHealthCheck", buildTimeConfig.healthEnabled);
```

Comment:
I don't know how it was done for the others but if it's not API, it should probably go in the `runtime.` subpackage.